ghpghx/models.py

```diff
@@ -0,0 +1,300 @@
+        default=162.3, validators=[MinValueValidator(10.0), MaxValueValidator(500.0)],
+        help_text="Mass density of the ground surrounding the borehole field [lb/ft^3]")
+    ground_specific_heat_btu_per_lb_f = models.FloatField(null=True, blank=True, 
+        default=0.211, validators=[MinValueValidator(0.01), MaxValueValidator(5.0)],
+        help_text="Specific heat of the ground surrounding the borehole field")
+    grout_thermal_conductivity_btu_per_hr_ft_f = models.FloatField(null=True, blank=True, 
+        default=1.0, validators=[MinValueValidator(0.01), MaxValueValidator(10.0)],
+        help_text="Thermal conductivity of the grout material in a borehole [Btu/(hr-ft-degF)]")
+    ghx_fluid_specific_heat_btu_per_lb_f = models.FloatField(null=True, blank=True, 
+        default=1.0, validators=[MinValueValidator(0.1), MaxValueValidator(10.0)],
+        help_text="Specific heat of the fluid in the GHX (nominally water) [Btu/(lb-degF)]")
+    ghx_fluid_mass_density_lb_per_ft3 = models.FloatField(null=True, blank=True, 
+        default=62.4, validators=[MinValueValidator(1.0), MaxValueValidator(200.0)],
+        help_text="Mass density of the fluid in the GHX (nominally water) [lb/ft^3]")
+    ghx_fluid_thermal_conductivity_btu_per_hr_ft_f = models.FloatField(null=True, blank=True, 
+        default=0.34, validators=[MinValueValidator(0.01), MaxValueValidator(5.0)],
+        help_text="Thermal conductivity of the fluid in the GHX (nominally water) [Btu/(hr-ft-degF)]")
+    ghx_fluid_dynamic_viscosity_lbm_per_ft_hr = models.FloatField(null=True, blank=True, 
+        default=2.75, validators=[MinValueValidator(0.1), MaxValueValidator(10.0)],
+        help_text="Dynamic viscosity of the fluid in the GHX (nominally water) [lb/(ft-hr)]")
+    ghx_fluid_flow_rate_gpm_per_ton = models.FloatField(null=True, blank=True, 
+        default=2.5, validators=[MinValueValidator(0.1), MaxValueValidator(10.0)],
+        help_text="Volumetric flow rate of the fluid in the GHX per peak ton heating/cooling [GPM/ton]")
+    ghx_pump_power_watt_per_gpm = models.FloatField(null=True, blank=True, 
+        default=15.0, validators=[MinValueValidator(0.0), MaxValueValidator(100.0)],
+        help_text="Pumping power required for a given volumetric flow rate of the fluid in the GHX [Watt/GPM]")
+    ghx_pump_min_speed_fraction = models.FloatField(null=True, blank=True, 
+        default=0.1, validators=[MinValueValidator(0.0), MaxValueValidator(1.0)],
+        help_text="The minimum turndown fraction of the pump. 1.0 is a constant speed pump.")
+    ghx_pump_power_exponent = models.FloatField(null=True, blank=True, 
+        default=2.2, validators=[MinValueValidator(0.1), MaxValueValidator(10.0)],
+        help_text="The pump power curve exponent")
+    max_eft_allowable_f = models.FloatField(null=True, blank=True, 
+        default=104.0, validators=[MinValueValidator(0.0), MaxValueValidator(150.0)],
```

Comment:
@Bill-Becker Any field that has a default value should not have `null=True` (which implies that a null value can be saved to the database). Please remove `null=True` from all values with defaults.

Reply:
Done